packages/cloudflare/src/cli/build/utils/workerd.ts

```diff
@@ -7,27 +7,67 @@ import logger from "@opennextjs/aws/logger.js";
 import { getCrossPlatformPathRegex } from "@opennextjs/aws/utils/regex.js";
 
 /**
- * Return whether the passed export map has the given condition
+ * This function transforms the exports (or imports) object from the package.json
+ * to only include the build condition if found (e.g. "workerd") and remove everything else.
+ * If no build condition is found, it keeps everything as is.
+ * It also returns a boolean indicating if the build condition was found.
+ * @param exports The exports (or imports) object from the package.json
+ * @param condition The build condition to look for
+ * @returns An object with the transformed exports and a boolean indicating if the build condition was found
  */
-export function hasBuildCondition(
-  exports: { [key: string]: unknown } | undefined,
-  condition: string
-): boolean {
-  if (!exports) {
-    return false;
-  }
+export function transformBuildCondition(exports: { [key: string]: unknown }, condition: string) {
+  const transformed: { [key: string]: unknown } = {};
+  const hasTopLevelBuildCondition = Object.keys(exports).some(
+    (key) => key === condition && typeof exports[key] === "string"
+  );
+  let hasBuildCondition = hasTopLevelBuildCondition;
   for (const [key, value] of Object.entries(exports)) {
     if (typeof value === "object" && value != null) {
-      if (hasBuildCondition(value as { [key: string]: unknown }, condition)) {
-        return true;
-      }
+      const { transformedExports, hasBuildCondition: innerBuildCondition } = transformBuildCondition(
+        value as { [key: string]: unknown },
+        condition
+      );
+      transformed[key] = transformedExports;
+      hasBuildCondition = hasBuildCondition || innerBuildCondition;
     } else {
-      if (key === condition) {
-        return true;
+      // If it doesn't have the build condition, we need to keep everything as is
+      // If it has the build condition, we need to keep only the build condition
+      // and remove everything else
+      if (!hasTopLevelBuildCondition) {
+        transformed[key] = value;
+      } else if (key === condition) {
+        transformed[key] = value;
       }
     }
   }
-  return false;
+  return { transformedExports: transformed, hasBuildCondition };
+}
+// We only care about these 2 fields
+interface PackageJson {
+  name: string;
+  exports?: { [key: string]: unknown };
+  imports?: { [key: string]: unknown };
+}
+
+/**
+ *
+ * @param json The package.json object
+ * @returns An object with the transformed package.json and a boolean indicating if the build condition was found
+ */
+export function transformPackageJson(json: PackageJson) {
+  const transformed: PackageJson = { ...json };
```

Comment:
`{...json}` does not really help here as we are updating objects -> `structureClone`?